pkg/generate/app/sourcelookup.go

```diff
@@ -15,6 +16,11 @@ import (
 	s2igit "github.com/openshift/source-to-image/pkg/scm/git"
 )
 
+var (
+	argumentGit         = regexp.MustCompile("^(http://|https://|git@|git://).*(?:#([a-zA-Z0-9]*))?$")
+	argumentGitProtocol = regexp.MustCompile("^(git@|git://)")
```

Comment:
but it is a good point that #foo is valid in a url, so this shouldn't be failng validation at all, even without special casing it.